### PR TITLE
C2LC-109: Use aria-disabled for the run button

### PR DESCRIPTION
**Pull request by chosww**

### CODE REVIEW

**Comment by simonbates** [comment]
@chosww I've added a commit to `master` to make `npm audit` not fail the build. Could you merge `master` into this branch please?

https://github.com/codelearncreate/c2lc-coding-environment/commit/fef4c3939ab2619026d4ed2ade6c3ba52f8c0041

**Comment by simonbates** [comment]
I've updated C2LC-109 to use a new `AriaDisablingButton` component for the run button. Could you please update this branch to use `AriaDisablingButton` for the Add and Delete editor buttons?

You will need to add some styling so that the buttons look disabled on hover and on active.

See: https://github.com/simonbates/c2lc-coding-environment/commit/88f342caa2554c4b1e404ebdcbc7f5ad36bb068a#diff-553ab764c97ca79a9fa98cd49952e754

**Comment by chosww** [comment]
> I've updated C2LC-109 to use a new `AriaDisablingButton` component for the run button. Could you please update this branch to use `AriaDisablingButton` for the Add and Delete editor buttons?
> 
> You will need to add some styling so that the buttons look disabled on hover and on active.
> 
> See: [simonbates@88f342c#diff-553ab764c97ca79a9fa98cd49952e754](https://github.com/simonbates/c2lc-coding-environment/commit/88f342caa2554c4b1e404ebdcbc7f5ad36bb068a#diff-553ab764c97ca79a9fa98cd49952e754)



> I've updated C2LC-109 to use a new `AriaDisablingButton` component for the run button. Could you please update this branch to use `AriaDisablingButton` for the Add and Delete editor buttons?
> 
> You will need to add some styling so that the buttons look disabled on hover and on active.
> 
> See: [simonbates@88f342c#diff-553ab764c97ca79a9fa98cd49952e754](https://github.com/simonbates/c2lc-coding-environment/commit/88f342caa2554c4b1e404ebdcbc7f5ad36bb068a#diff-553ab764c97ca79a9fa98cd49952e754)

Sure, I will do that now 👍

**Comment by chosww** [comment]
@simonbates should we move disabled styling to the global css file? Making disabled class for all the button components seems redundant.

**Comment by simonbates** [comment]
Yes, I think having a general disabled button class with the rules in the global css file is a good idea. Thanks.

File: src/AriaDisablingButton.js [hunk 1]
```diff
@@ -8,7 +8,6 @@ type AriaDisablingButtonProps = {
     onClick: (evt: any) => void,
     disabled: boolean,
     className?: string,
-    disabledClassName: string,
```
Comment (simonbates) [inline src/AriaDisablingButton.js]: Please put this property back. We can still use a general CSS class in our global styles but provide it to the AriaDisabingButton instances as a property. This decouples the CSS class specifics from the component implementation.

Comment (chosww) [reply src/AriaDisablingButton.js]: My bad, will do it now :)

File: src/index.css [hunk 2]
```diff
@@ -42,6 +42,21 @@ button.command-block--pressed:hover {
     border: 2px dashed #000000;
 }
 
+button.btn.button--disabled svg {
+    stroke: #777;
+    fill: #777;
+}
+
+button.btn.button--disabled {
```
Comment (simonbates) [inline src/index.css]: The #FFF background-color is specific to `btn-primary`. Change the selector to: `button.btn.button--disabled`.

File: src/index.css [hunk 3]
```diff
@@ -42,6 +42,21 @@ button.command-block--pressed:hover {
...
+    background-color: #FFF;
+    cursor: default;
+}
+
+button.btn.button--disabled:active:focus {
```
Comment (simonbates) [inline src/index.css]: The #FFF background-color is specific to `btn-primary`. Change the selector to: `button.btn-primary.button--disabled:active:focus`.

**Pull request merged**